client/storage_accounts.go

```diff
@@ -0,0 +1,206 @@
+		params   = query.Params{ApiVersion: "2022-05-01", StatusOnly: statusOnly}.AsMap()
+		headers  map[string]string
+		response azure.StorageAccountList
+	)
+	if res, err := s.resourceManager.Get(ctx, path, params, headers); err != nil {
+		return response, err
+	} else if err := rest.Decode(res.Body, &response); err != nil {
+		return response, err
+	} else {
+		return response, nil
+	}
+}
+
+func (s *azureClient) ListAzureStorageAccounts(ctx context.Context, subscriptionId string, statusOnly bool) <-chan azure.StorageAccountResult {
+	out := make(chan azure.StorageAccountResult)
+
+	go func() {
+		defer close(out)
+
+		var (
+			errResult = azure.StorageAccountResult{
+				SubscriptionId: subscriptionId,
+			}
+			nextLink string
+		)
+
+		if result, err := s.GetAzureStorageAccounts(ctx, subscriptionId, statusOnly); err != nil {
+			errResult.Error = err
+			out <- errResult
+		} else {
+			for _, u := range result.Value {
+				out <- azure.StorageAccountResult{SubscriptionId: subscriptionId, Ok: u}
+			}
+
+			nextLink = result.NextLink
+			for nextLink != "" {
+				var list azure.StorageAccountList
+				if url, err := url.Parse(nextLink); err != nil {
+					errResult.Error = err
+					out <- errResult
+					nextLink = ""
+				} else if req, err := rest.NewRequest(ctx, "GET", url, nil, nil, nil); err != nil {
+					errResult.Error = err
+					out <- errResult
+					nextLink = ""
+				} else if res, err := s.resourceManager.Send(req); err != nil {
+					errResult.Error = err
+					out <- errResult
+					nextLink = ""
+				} else if err := rest.Decode(res.Body, &list); err != nil {
+					errResult.Error = err
+					out <- errResult
+					nextLink = ""
+				} else {
+					for _, u := range list.Value {
+						out <- azure.StorageAccountResult{
+							SubscriptionId: "/subscriptions/" + subscriptionId,
+							Ok:             u,
+						}
+					}
+					nextLink = list.NextLink
+				}
+			}
+		}
+	}()
+	return out
+}
+
+// ==
+// Storage containers
+// ==
+
+func (s *azureClient) GetAzureStorageContainer(ctx context.Context, subscriptionId, groupName, saName, scName, expand string) (*azure.StorageContainer, error) {
+	var (
+		path     = fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s/blobServices/default/containers/%s", subscriptionId, groupName, saName, scName)
+		params   = query.Params{ApiVersion: "2022-05-01", Expand: expand}.AsMap()
+		headers  map[string]string
+		response azure.StorageContainer
+	)
+	if res, err := s.resourceManager.Get(ctx, path, params, headers); err != nil {
+		return nil, err
+	} else if err := rest.Decode(res.Body, &response); err != nil {
+		return nil, err
+	} else {
+		return &response, nil
+	}
+}
+
+func (s *azureClient) GetAzureStorageContainers(ctx context.Context, subscriptionId string, resourceGroupName string, saName string, statusOnly bool) (azure.StorageContainerList, error) {
```

Comment:
We should expose all the URI params in this API. Refer to: https://docs.microsoft.com/en-us/rest/api/storagerp/blob-containers/list?tabs=HTTP#uri-parameters

Reply:
Added all 3 new ones as strings(along with tests and all implementations edited), however in testing I could not get the filter working, unless it's empty. I get
```
"code": "InvalidRequestParameters",
    "message": "Request parameters are invalid: $filter."
```
when trying with both a simple string or an expresion like `name eq bla`. I'll mark it as resolved since the only use has an empty filter, but I'd be curious to know if you or someone else got this filter working in the REST API.